google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -0,0 +1,145 @@
+        if not experiment:
+            raise ValueError(f"Invalid experiment {experiment}.")
+
+        store = _MetadataStore.get()
+        if not store:
+            logging.info(
+                f"Creating a default MetadataStore for experiment {experiment}"
+            )
+            _MetadataStore.create()
+
+        context = _Context.get(resource_name=experiment)
+        if not context:
+            logging.info(f"Creating a Context for experiment {experiment}")
+            context = _Context.create(
+                resource_id=experiment,
+                schema_title="system.Experiment",
+                schema_version="0.0.1",
+            )
+        self._experiment = context.name
+
+    def set_run(self, run: str):
+        if not self._experiment:
+            raise ValueError(
+                "No experiment set for this run. Make sure to call aiplatform.init(experiment='my-experiment') or "
+                "aiplatform.set_experiment(experiment='my-experiment') before trying to set_run. "
+            )
+        if not run:
+            raise ValueError(f"Invalid run {run}.")
+
+        execution = _Execution.get(resource_name=run)
+        if not execution:
+            logging.info(f"Creating an Execution for run {run}")
+            execution = _Execution.create(
+                resource_id=run, schema_title="system.Run", schema_version="0.0.1",
+            )
+        self._run = execution.name
+
+    def log_param(self, name: str, value):
+        if not isinstance(value, (float, int, str)):
+            raise ValueError(
+                f"Invalid type for parameter value {value}. It should be one of [int, float, str]"
+            )
+        return self.log_params({name: value})
+
+    def log_params(self, params: Dict):
+        if not self._experiment:
+            raise ValueError(
+                "No experiment set for logging parameters. Make sure to call aiplatform.init("
+                "experiment='my-experiment') or aiplatform.set_experiment(experiment='my-experiment') before trying "
+                "to log_params. "
+            )
+        if not self._run:
+            raise ValueError(
+                "No run set for logging parameters. Make sure to call aiplatform.init(experiment='my-experiment', "
+                "run='my-run') or aiplatform.set_run('my-run') before trying to log_params. "
+            )
+        execution = _Execution.get(resource_name=self._run)
+        if not execution:
+            logging.info(f"Creating an Execution for run {self._run}")
+            execution = _Execution.create(
+                resource_id=self._run,
+                schema_title="system.Run",
+                schema_version="0.0.1",
+                metadata=params,
+            )
+        else:
+            logging.info(f"Updating Execution for run {self._run}")
+            execution.update(metadata=params)
+        self._run = execution.name
+
+    def log_metric(self, name: str, value):
```

Comment:
Value type should be `Union[str, float, int]`

Reply:
Thanks!